backend/app/assets/javascripts/spree/backend/admin.js.erb

```diff
@@ -7,6 +7,7 @@
 //= require spree/backend/product_picker
 //= require spree/backend/option_value_picker
 //= require spree/backend/taxons
+//= require spree/backend/country_states
```

Comment:
I removed the old update_state method from the backend and replaced all with the new javascript.
The old method did not checked if the country needed a state, so a text field was always shown. This method does.